modules/photometric_calib/src/Reader.cpp

```diff
@@ -0,0 +1,81 @@
+//
+// Created by Nan Yang on 17/6/14.
+//
+
+#include "precomp.hpp"
+#include "opencv2/photometric_calib/Reader.hpp"
+
+namespace cv { namespace photometric_calib{
+
+unsigned long Reader::getNumImages()
+{
+    return (unsigned long)files.size();
+}
+
+void Reader::loadTimestamps(std::string timesFile)
+{
+    std::ifstream timesStream;
+    timesStream.open(timesFile.c_str());
+    timeStamps.clear();
+    exposureTimes.clear();
+    while (!timesStream.eof() && timesStream.good())
+    {
+        char buf[1000];
+        timesStream.getline(buf, 1000);
+
+        int id = 0;
+        double timeStamp = 0.0;
+        float exposureTime = 0.0;
+
+        CV_Assert(3 == scanf(buf, "%d %lf %f", &id, &timeStamp, &exposureTime));
+
+        timeStamps.push_back(timeStamp);
+        exposureTimes.push_back(exposureTime);
+    }
+    timesStream.close();
+
+    CV_Assert(timeStamps.size() == getNumImages() && exposureTimes.size() == getNumImages());
+}
+
+Reader::Reader(std::string folderPath, std::string timesPath)
+{
+    String cvFolderPath(folderPath);
+    glob(cvFolderPath, files);
+    CV_Assert(files.size() > 0);
+    std::sort(files.begin(), files.end());
+    loadTimestamps(timesPath);
+
+    width = 0;
+    height = 0;
+
+    for(unsigned long i = 0; i < files.size(); ++i)
+    {
+        Mat img = imread(files[i]);
+        CV_Assert(img.type() == CV_8U);
+        if(0 == i)
+        {
+            width = img.cols;
+            height = img.rows;
+        }
+        else
+        {
+            CV_Assert(width == img.cols && height == img.rows);
+        }
+    }
+
+    std::cout<<getNumImages()<<" imgases from"<<folderPath<<" loaded successfully!"<<std::endl;
```

Comment:
Hi Vincent, thanks a lot for your comments!

Can I use **std::cerr** to output some error info?